lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -589,7 +591,11 @@ export class UserAgentApplication {
      *
      */
     acquireTokenSilent(userRequest: AuthenticationParameters): Promise<AuthResponse> {
-
+        const requestCorrelationId = userRequest.correlationId || uuid.v4();
+        const apiEvent: ApiEvent = new ApiEvent(requestCorrelationId, this.logger);
+        apiEvent.apiEventIdentifier = API_EVENT_IDENTIFIER.AcquireTokenSilent;
+        apiEvent.apiCode = API_CODE.AcquireTokenSilent;
```

Comment:
Is it possible to flatten this to a single constructor call? Having public members set without getters or setters is a bad code smell IMO.

Reply:
I have thought about that, I am happy to revisit the design when we instrument more, but probably not in this pr

Reply:
On a side note, we also worked very hard to simplify this API :-), so it helps instrumenting the code in independent helpers.